### PR TITLE
Add a note about the `test` package not being fully migrated

### PR DESCRIPTION
**Pull request by jakemac53**
cc @mit-mit

### CODE REVIEW

File: src/null-safety/index.md [hunk 1]
```diff
@@ -344,6 +344,12 @@ The Dart team is currently aware of the following issues:
   * Migration of the pub.dev packages owned by the Dart team
     is nearly complete, but a few are still missing. See pub.dev for
     [null-safe packages from the Dart team][ns-dart-pkgs].
+  * When publishing packages with a regular (non-dev) dependency on
```
Comment (mit-mit) [inline src/null-safety/index.md:347]: I wonder if we need to start with a sentence that explains that direct dependencies are not typical? Maybe something like:

Packages which have unit tests typically have a dev dependency on `package:test`. A few packages extend the framework in `package:test`, and may have a direct dependency. When publishing those packages...

File: src/null-safety/index.md [hunk 2]
```diff
@@ -344,6 +344,12 @@ The Dart team is currently aware of the following issues:
...
+    the `test` package you may get pub warnings about it not being
+    fully migrated. As long as you are not ignoring any of the
+    `import_of_legacy_library_into_null_safe` or
```
Comment (mit-mit) [inline src/null-safety/index.md:350]: Can we add links to those lints?

Comment (jakemac53) [reply src/null-safety/index.md:350]: I would if I had any idea where to link to :)

File: src/null-safety/index.md (hunk 1 above)
Comment (jakemac53) [reply src/null-safety/index.md]: done

**Review by kwalrath — COMMENTED** [review]
LGTM with minor wording changes

File: src/null-safety/index.md (hunk 2 above)
Comment (kwalrath) [reply src/null-safety/index.md]: I'll get this...

Comment (parlough) [reply src/null-safety/index.md]: As a reference I don't know if there is currently on for `import_of_legacy_library_into_null_safe` but there is one for [`export_legacy_symbol`](https://dart.dev/tools/diagnostic-messages#export_legacy_symbol).

File: src/null-safety/index.md [hunk 3]
```diff
@@ -344,6 +344,18 @@ The Dart team is currently aware of the following issues:
...
+  * Packages that have unit tests typically have
+    a dev dependency on the [`test` package][`test`].
+    A few packages extend the framework in package:test,
+    and might have a direct dependency on it.
+    When publishing these packages you might get pub warnings about
+    the `test` package not being fully migrated.
+    As long as your package doesn't ignore any of the
+    `import_of_legacy_library_into_null_safe` or
+    `export_legacy_symbol` analyzer warnings,
+    then you can ignore this warning and safely publish your package.
```
Comment (kwalrath) [inline src/null-safety/index.md:356]:
```suggestion
    [`export_legacy_symbol`][] analyzer warnings,
    then you can ignore this warning and safely publish your package.
```

File: src/null-safety/index.md [hunk 4]
```diff
@@ -344,6 +344,18 @@ The Dart team is currently aware of the following issues:
...
+
+[`test`]: {{site.pub-pkg/test}}
```
Comment (kwalrath) [inline src/null-safety/index.md:358]:
```suggestion
[`export_legacy_symbol`]: /tools/diagnostic-messages#export_legacy_symbol
[`test`]: {{site.pub-pkg/test}}
```

**Review by kwalrath — APPROVED** [review]

**Review by kwalrath — CHANGES_REQUESTED** [review]
Oops, I need to fix the formatting before we merging this

**Review by kwalrath — APPROVED** [review]

**Pull request merged**